utils/buildcli/buildcli/clang.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import absolute_import
+
+from collections import namedtuple
+import platform
+import subprocess
+
+from . import xcrun
+from .which import which
+
+# host_clang() prefers versions of Clang in this order.
+# Note that this array must be updated when new, major versions of Clang
+# are released.
+_CLANG_SUFFIXES = ['38', '37', '36', '35']
+
+
+class CompilerExecutable(namedtuple('CompilerExecutable', 'cc cxx')):
+    """
+    A named tuple consisting of two paths:
+
+    1. 'cc' is the path to a program used for compiling C.
+    2. 'cxx' is the path to a program used for compiling C++.
+    """
+    @property
+    def is_valid(self):
+        """
+        Return a boolean value indicating whether this instance represents
+        a valid C and C++ compiler pair.
+        """
+        return self.cc is not None and self.cxx is not None
+
+
+def _freebsd_release_date():
+    """
+    Return the release date for FreeBSD operating system on this host.
+    If the release date cannot be ascertained, return None.
```

Comment:
I think it will return `0` instead, or sometimes throw an error, if `sysctl` returns a non-integer.

I'd just halt the execution at that point.  There is no reasonable way a caller could recover from an error in this function.  They either need the version and they call this function, or they don't, and they shouldn't be calling it.